testsuite/features/support/env.rb

```diff
@@ -130,26 +135,80 @@ def capybara_register_driver
 # Define the current feature scope
 Before do |scenario|
   $feature_scope = scenario.location.file.split(%r{(\.feature|/)})[-2]
+  if $timeout_failure_count >= $timeout_threshold
+    warn "Timeout failure threshold reached, stopping test suite."
+    exit(1)
+  end
+  feature_path = scenario.location.file
+  $feature_filename = feature_path.split(%r{(\.feature|/)})[-2]
+  next if get_context('user_created') == true
+
+  # Create own user based on feature filename. Exclude core, reposync, finishing and build_validation features.
+  unless feature_path.match?(/core|reposync|finishing|build_validation/)
+    step %(I create a user with name "#{$feature_filename}" and password "linux")
+    add_context('user_created', true)
+  end
```

Comment:
We are duplicating these steps in another Before method, please let's remove this from this Before method. We better have separated Before methods each of them having a single responsibility, it gonna be more readable and easier to maintain.